### PR TITLE
fixes #21193 Expose a number of GraphStage attributes

### LINKED ISSUE #21193: Expose a number of GraphStage attributes
GraphStages which are not case classes should have there attributes accessible for external materializers that need to access them. Currently these are TickSource and Throttle. (A few others may be included as well).

### PR DESCRIPTION
**Pull request by kkasravi**

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

**Comment by ktoso** [comment]
LGTM, thanks :)

File: akka-stream/src/main/scala/akka/stream/impl/Throttle.scala [hunk 1]
```diff
@@ -15,11 +15,11 @@ import scala.concurrent.duration.{ FiniteDuration, _ }
  * INTERNAL API
  */
 class Throttle[T](
```
Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/impl/Throttle.scala]: final

**Comment by drewhk** [comment]
LGTM

**Comment by patriknw** [comment]
LGTM

**Comment by patriknw** [comment]
Refs #21193

**Comment by kkasravi** [comment]
Thanks @patriknw !

**Pull request merged**